scene/gui/color_picker.cpp

```diff
@@ -607,10 +616,18 @@ void ColorPicker::set_picker_shape(PickerShapeType p_shape) {
 	if (p_shape == current_shape) {
 		return;
 	}
-	shape_popup->set_item_checked(current_shape, false);
-	shape_popup->set_item_checked(p_shape, true);
+	if (current_shape != SHAPE_NONE) {
+		shape_popup->set_item_checked(current_shape, false);
+	}
+	if (p_shape != SHAPE_NONE) {
+		shape_popup->set_item_checked(p_shape, true);
+	}
 
-	btn_shape->set_icon(shape_popup->get_item_icon(p_shape));
+	if (p_shape == SHAPE_NONE) {
+		btn_shape->set_icon(shape_popup->get_item_icon(SHAPE_HSV_RECTANGLE));
+	} else {
+		btn_shape->set_icon(shape_popup->get_item_icon(p_shape));
+	}
```

Comment:
```suggestion
	if (p_shape != SHAPE_NONE) {
		btn_shape->set_icon(shape_popup->get_item_icon(p_shape));
	}
```
Simpler. Icon is irrelevant if the button is invisible.